enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/roles/Candidate.java

```diff
@@ -38,124 +38,178 @@
     @Override
     public Outcome handle( RaftMessages.RaftMessage message, ReadableRaftState ctx, Log log ) throws IOException
     {
-        Outcome outcome = new Outcome( CANDIDATE, ctx );
+        return message.visit( new Visitor( ctx, log ) );
+    }
 
-        switch ( message.type() )
-        {
-            case HEARTBEAT:
-            {
-                RaftMessages.Heartbeat req = (RaftMessages.Heartbeat) message;
+    private static class Visitor implements RaftMessages.Visitor<Outcome, IOException>
```

Comment:
I wonder if this static inner class could be merged with the containing class.

Reply:
ReadableRaftState and Log are constructor parameters for the inner visitor class. If it weren't a class they would have to be passed into the dispatch method, which would make the visitor interface less general or rather more complex.